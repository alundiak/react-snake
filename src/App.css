/* taken from https://gridbyexample.com/examples/example18/ */
body {
  margin: 40px;
}

.App {
  text-align: center;
}
