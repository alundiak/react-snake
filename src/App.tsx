import React from 'react';
import { SnakeGame } from './components/SnakeGame';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="">Snake</header>
      <main>
        <SnakeGame />
      </main>
    </div>
  );
}

export default App;
