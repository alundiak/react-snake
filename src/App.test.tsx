// import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

// Look details here
// https://github.com/vitest-dev/vitest/blob/main/examples/react/test/basic.test.tsx

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('renders component', () => {
  const component = renderer.create(<App />);

  const tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // const linkElement = screen.getByText(/Snake/i);
  // expect(linkElement).toBeInTheDocument();
});
