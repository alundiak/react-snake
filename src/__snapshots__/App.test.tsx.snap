// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders component 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h2>
      Snake
    </h2>
  </header>
  <main>
    <div
      className="light"
    >
      <div
        className="wrapper"
      >
        <div
          className="point   light"
          onClick={[Function]}
        >
          0
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y1
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y2
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y3
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y4
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y5
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y6
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          y7
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        >
          x1
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x2
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point target  light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x3
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x4
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point  snake head light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x5
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x6
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        >
          x7
        </div>
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
        <div
          className="point   light"
          onClick={[Function]}
        />
      </div>
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        Use 
        dark
         mode
      </label>
    </div>
  </main>
</div>
`;
